File: Search a 2D Matrix/Search a 2D Matrix.py
class Solution:
    # @param matrix, a list of lists of integers
    # @param target, an integer
    # @return a boolean
    def searchMatrix(self, matrix, target):
        m=0
        while target>matrix[m][0]and m!=len(matrix)-1:
            m+=1
        if target<matrix[m][0] and m!=0:
            m-=1
        for i in range(len(matrix[m])):
            if target==matrix[m][i]:
                return True
        return False

File: Search a 2D Matrix/Search a 2D Matrix.java
public class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
       int index=0;
       while(target>matrix[index][0]&&index!=matrix.length-1){
           index++;
       }
       if(target<matrix[index][0]&&index!=0){
           index--;
       }
       for(int i=0;i<matrix[index].length;i++){
           if(target==matrix[index][i]) return true;
       }
       return false;
    }
}
